src/crypto/tls/common.go

```diff
@@ -674,6 +717,30 @@ type Config struct {
 	// used for debugging.
 	KeyLogWriter io.Writer
 
+	// ClientECHConfigs are the parameters used by the client when it offers the
+	// ECH extension. If a suitable configuration is found and the client
+	// supports TLS 1.3, then ECH will be offered in this handshake.
+	ClientECHConfigs []ech.Config
```

Comment:
This should really be a single config. If the developer fails to provide a proper config, then they should receive an error. Why are multiple values supported?

Reply:
The server might offer multiple versions of ECH. The version supported by the client is determined by a code point defined internally (i.e., extensionECH in common.go).